### DISCUSSION TITLE
Helping on running Angular sandbox

### DISCUSSION THREAD

**Discussion by ryan-bendel** [storybookjs/storybook#28850] [Help] [answered]
### Summary

Hi all,

I have a fork checked out, ran `yarn start` from root, `cd` into `code` as per docs, and then ran `yarn task --prod` 

From the interactive terminal I choose:
`Run the sandbox in development mode (dev)`
`angular-cli/default-ts`

Also tried `yarn build --prod --watch angular core addon-docs` instead of the above interactive version, which _seems_ to compile, but the loaded Storybook doesn't show the angular sandbox, only the default react/vite one

But it crashes with lots of Typescript compilation errors.

Maybe i'm running things the wrong way?

I've made sure i've got a clean workspace (removed any sandboxes/dist folders/node_modules prior to running)

This is on node v22.6.0

Any help is appreciated! 🙏

**Comment by Marklb** [comment, marked as answer]
I don't think the website docs mention that prod needs to be used for Angular, but maybe it should. Either way, I assume you are looking at `CONTRIBUTING.md`, based on the snippets that you mentioned, which is where you should have looked for the right answer.

My explanation got a little long, so I will just summarize the answer, if you don't need the details. Basically, you just want to run `yarn task --prod` and pick `... (compile)`. If working on Angular, all `yarn task` and `yarn build` commands need to be using `--prod`.

You could run `yarn task --task dev --template angular-cli/default-ts --prod --start-from=compile` to recompile and start the angular sandbox, but you probably don't want to add `--start-from=compile` each time because it may interfere with `yarn build --prod --watch angular` if you have it running.

Explanation:

I think you are getting the errors, because of the order that you ran the commands. If you are just wanting to run the Angular sandbox then you don't need to worry about `yarn start`. The docs mention to run `yarn start`, because it will do all the setup to get you started with a sandbox and the default is react-vite-default-ts. The problem is Angular's sandboxes don't build with Storybook's dev Typescript config. When you ran `yarn start` it compiled without `--prod`. When you ran `yarn build --prod --watch angular core addon-docs` that would have rebuilt `angular`, `core` and `addon-docs` in prod mode, but the other packages are still not built in prod mode. *If you are going to just be working on Angular then you probably only need `yarn build --prod --watch angular`, but it just depends what code you will be changing.`

The fix:
https://github.com/storybookjs/storybook/blob/edd0fd56aa64d1b94db79be7df0e19ad9a447818/CONTRIBUTING.md?plain=1#L71-L77
The important part that may not be obvious is

> ..you will need to append `--prod` to the above mentioned commands..

The problem is that you ran them without `--prod`, initially, so you could rerun them with it (`yarn start --prod`) or just do the important part and run `yarn task --prod` and choose `... (compile)`. After that, you can `cd code; yarn build --prod --watch angular <other packages you want to change>` then `yarn task --prod` and pick the angular sandbox.

**Reply by ryan-bendel** [reply]
Ah perfect, thanks for the detailed answer @Marklb - I kinda/sorta worked around it in the end by going down the path of "Build the static version of the sandbox (build)" after the `yarn task --prod` - took a bit longer as that's building the static version 😆 

I did indeed follow the `Making code changes when working on Angular-specific code` of the `CONTRIBUTING.md` 👍  but will def save these instructions for my next contribution

Thanks again!
